File: create_admin.py
import requests
import sys

def create_admin():
    url = "http://localhost:5000/api/auth/register"
    
    # Default admin credentials
    data = {
        "name": "Admin User",
        "username": "admin",
        "password": "admin123",
        "role": "Admin"
    }
    
    try:
        response = requests.post(url, json=data)
        
        if response.status_code == 201:
            print("✅ Admin user created successfully!")
            print("\nLogin Credentials:")
            print("Username: admin")
            print("Password: admin123")
            print("\n⚠️  Please change the password after first login!")
        else:
            print(f"❌ Error: {response.json().get('message', 'Failed to create admin')}")
            
    except requests.exceptions.ConnectionError:
        print("❌ Error: Cannot connect to backend server")
        print("Make sure the Flask server is running on http://localhost:5000")
        sys.exit(1)
    except Exception as e:
        print(f"❌ Error: {str(e)}")
        sys.exit(1)

if __name__ == "__main__":
    print("Creating admin user...")
    create_admin()

File: backend/createAdmin.js
const mongoose = require('mongoose');
const bcrypt = require('bcryptjs');
const User = require('./models/User');
require('dotenv').config();

async function createAdmin() {
  try {
    // Connect to MongoDB
    await mongoose.connect(process.env.MONGODB_URI);
    console.log('Connected to MongoDB');

    // Check if admin exists
    const existingAdmin = await User.findOne({ username: 'admin' });
    if (existingAdmin) {
      console.log('❌ Admin user already exists!');
      process.exit(0);
    }

    // Hash password
    const hashedPassword = await bcrypt.hash('admin123', 10);

    // Create admin user
    const admin = new User({
      name: 'Admin User',
      username: 'admin',
      password: hashedPassword,
      role: 'Admin'
    });

    await admin.save();

    console.log('✅ Admin user created successfully!');
    console.log('\nLogin Credentials:');
    console.log('Username: admin');
    console.log('Password: admin123');
    console.log('\n⚠️  Please change the password after first login!');

    process.exit(0);
  } catch (error) {
    console.error('❌ Error:', error.message);
    process.exit(1);
  }
}

createAdmin();
